### ISSUE TITLE
Loop stops working after extended period of phone screen being off (eg 30 mins)

### ISSUE THREAD

**Issue by chookchook1** [LoopKit/Loop#1230]
Loop V2
Iphone 8
iOS 13.3
Dexcom G5
Background App Refresh ON 

When  iphone screen is off Loop stops looping. It is hard to get a screenshot as waking phone can trigger a new loop. It appears that the loop stopping is due to it not using the current glucose, even though it is current on CGM source. Screen shot shows HUD with old glucose, while CGM source is current. Not a huge difference in time here, but have seen this repeatedly over larger time frames. Waking phone resolves this issue.  At night phone screen can be left on, but during day has undesirable risks such as accidental carb entries.

A facebook user has reported similar issues, however link to their post is not working properly here. 

![fullsizeoutput_2541](https://user-images.githubusercontent.com/20076960/71935973-45282580-31fc-11ea-9b86-2fee028557e0.jpeg)
![fullsizeoutput_253f](https://user-images.githubusercontent.com/20076960/71935989-5113e780-31fc-11ea-8dfc-13ed8261083f.jpeg)

**Comment by Kdisimone** [comment]
There have been sporadic reports of people having phones that will not loop at all when the phone is not actively open and awake. 

The only thing that seems to resolve it reliably is to either get a new phone (undesirable, I understand) or wipe the phone and set it up as a new device. That also fixes the problem. Bit of a pain in the butt, but it does fix whatever iOS bug is causing the problem.

**Comment by chookchook1** [comment]
> There have been sporadic reports of people having phones that will not loop at all when the phone is not actively open and awake.
> 
> The only thing that seems to resolve it reliably is to either get a new phone (undesirable, I understand) or wipe the phone and set it up as a new device. That also fixes the problem. Bit of a pain in the butt, but it does fix whatever iOS bug is causing the problem.

Thanks, we will try that. I have several other issues (no blue suggestion and missing cab and insulin summary from bolus pop up for example) that I was planning on logging here as well- would they be likely to be resolved by this (ie should I hold off on logging them ?!)

**Comment by Kdisimone** [comment]
Yes. Hold off. Those symptoms sound like your Loop is slow to load. Likely tied to same issue.

**Comment by mswannee** [comment]
Hi Katie, 
Hamish posted this for me. I will raise new issues once I've done the master reboot of my phone and see if this fixes all the issues I've noticed. Can't do it for a couple of weeks as going on road trip, so will have to grin and bear the issues. Thanks for the advice. 
Regards,
Mike

**Comment by chookchook1** [comment]
> Hi Katie,
> Hamish posted this for me. I will raise new issues once I've done the master reboot of my phone and see if this fixes all the issues I've noticed. Can't do it for a couple of weeks as going on road trip, so will have to grin and bear the issues. Thanks for the advice.
> Regards,
> Mike

Just to be clear, we have had the same issues, so there are 2 different users with the problem. We hope to be able to do the reset soon and will report back.

**Comment by mswannee** [comment]
Hi again, was getting very frustrated with a BSL of 13mmol after lunch today with no Bolus advised after entering 45gm carbs. BSL was right in range beforehand. So as a result and the fact I'm on night shift tonight (before heading off on road trip) I bit the bullet and decided to rebuild phone from scratch then Loop using previously downloaded code for V2.  Backed up iPhone first. Erased all data. And set up from scratch and then from saved backup. Then rebuilt Loop V2. 
First test and only test so far was entering 15gm carb and straight away 2unit of insulin Bolus advised. Looks promising. I'll update more later as haven't left phone alone for to long yet. Stay tuned. One thing to note was that all my Loop settings stayed when V2 built. Didn't need to update them. Thanks again. 
![ED7C6F75-BFE9-4B0D-9900-64FD1697625D](https://user-images.githubusercontent.com/37945901/71961914-821c0880-324c-11ea-8623-9139989e689d.jpeg)

**Comment by mswannee** [comment]
So since rebuilding iOS 13.3 on iPhone 8+, reloading backup, rebuilding Loop V2 I've had assume events on 'Loop inactive for 20min'![Image](https://i.imgur.com/bXDl8ro.jpg)![Image](https://i.imgur.com/9Ntqryr.jpg)

**Comment by live4sw** [comment]
What error comes up when you tap the red loop? Does a test command from your RL work?

**Comment by mswannee** [comment]
That's the strange thing, NOTHING.

**Comment by mswannee** [comment]
I saved a report I can add that when I figure out how to.

**Comment by mswannee** [comment]
Haven't tried the RL link test. It does seem to fix itself after opening Loop for a little bit of time. Personally it reminds me of the issues I was having after iOS 12 and Apple closing apps running in the background. I do have background so refresh ON.

**Comment by mswannee** [comment]
Here's a snapshot just now of Bolus going with Amber loop. ![Image](https://i.imgur.com/wB6LB4c.jpg)

**Comment by mswannee** [comment]
At least Bolus suggestion is coming up straight away. One problem fixed.

**Comment by mswannee** [comment]
Loop Report 2020-01-08 21_33_55+11_00.md

**Comment by mswannee** [comment]
To Whom It May Concern,
Here is a pdf of my last reports after rebuild iphone and Loop V2. This was after having 2 occurences of Loop inactive for 20mins. 
[Loop Report 2020-01-08 21_33_55+11_00.pdf](https://github.com/LoopKit/Loop/files/4035436/Loop.Report.2020-01-08.21_33_55%2B11_00.pdf)

**Comment by mswannee** [comment]
And here is latest snapshot of CGM being over 50mins old but Loop still working.![Image](https://i.imgur.com/Xsfpgfc.jpg)

**Comment by Kdisimone** [comment]
@mswannee your loop doesn't appear to be working.  See the insulin delivery graphs. Please reset your pump time using the Change Time command in the RileyLink menu. Do not use the pump menus to change time.

**Comment by mswannee** [comment]
Thanx @Kdisimone. Done that just now. It was strange that when I relaoded V2 after rebuilding iPhone that all my settings for Loop were already populated. I was expecting to have to relaod those. I'll advise if no more "Loop inactive" errors

**Comment by mswannee** [comment]
Well, my issues may be fixed. I have not had any further reports of 'Loop Inactive' for the last 3 hrs since Resetting Pump Clock via RL menu (instead of Pump Menu). After adding Carbs my Boluses have been suggested as required.
Thans again @Kdisimone for your help. @chookchook1 may still have issues so I guess we await his feedback on his iPhone Reboot/Loop Rebuild.

**Comment by timmy354321** [comment]
I am also having a similar issue. When I open the app sometimes the recent glucose data loads in a few seconds. Sometimes the CGM data doesn’t load and the loop app crashes. I was reading the comments above and I don’t seem to have a command to update the time within the rileylink screen. I’m using an iPhone 5 with 12.4.4. This is a new issue with V2.0. The Omnipod version prior to this never crashed or lagged. I’m attaching the report as well. 
![BA620476-FC6A-404E-A854-9EE6E248A562](https://user-images.githubusercontent.com/48967293/72165524-46d52300-338d-11ea-9c91-deaacd0b60c1.png)
[LoopIssueReport.pdf](https://github.com/LoopKit/Loop/files/4046422/LoopIssueReport.pdf)

**Comment by chookchook1** [comment]
This issue has resolved for us. We rebuilt and set up a phone from new. (for future reference

**Comment by Kdisimone** [comment]
Great! Glad to hear all resolved!! Thanks for letting  us know.

**Comment by doubleii111** [comment]
I've been having the same issue. So far I've turned off Nightscout in Loop, I had disabled Dexcom Share a while ago, I don't have Loop or Dex on my watch, and last night I disabled Share in the Dex app. Every time I make a change it stabilizes for a day or two and then re-occurs.

I have been avoiding resetting my phone thus far but fear it may be the next step. Is there anything else I can try to avoid it? I've been wondering if my auto lock timer is too quick and that is what is causing Loop app to get confused/bogged down?

Iphone 11, iOS 13.3, Loop 2.0. Dex G6 and Omnipod. Background app refresh is ON.

**Comment by live4sw** [comment]
Did you try rebuilding the app itself (without deleting it)? Also, when your loop stops working and you check settings, does your RileyLink have both firmwares?

**Comment by doubleii111** [comment]
Not yet. If I leave it for a while it generally restores, but I have to reboot Loop app a bunch of times first. It frequently loses the CGM data when it happens, yet the Dex app is fine. I can also usually bolus directly and send test beeps while the loop icon is red.

Forgot to say I've also cleaned out the mLab cache before I got to the point of removing Nightscout.

I'll check the firmwares, haven't noticed that. Thanks.
